tests/lax_numpy_test.py

```diff
@@ -849,8 +849,9 @@ def np_fun(x):
         "np_op": getattr(np, rec.name), "jnp_op": getattr(jnp, rec.name),
         "initial": initial, "axis": axis, "keepdims": keepdims, "inexact": rec.inexact}
         for shape in rec.shapes for dtype in rec.dtypes
-        for axis in list(range(-len(shape), len(shape))) + [None]
-        for initial in [0, 1] for keepdims in [False, True]
+        for initial in [None, 0, 1] for keepdims in [False, True]
+        for axis in (list(a for a in range(-len(shape), len(shape)) if shape[a] >= 1 or initial is not None)
+            + ([] if initial is None else [None]))
```

Comment:
I've been staring at this list comprehension for a while, and it's still not clear to me the logic behind it. When things get this complicated, I'd advocate for refactoring...

This parameterization is for `testReducerInitial`, so it seems strange to add `initial=None`. Perhaps it would be cleaner to add another test case for the coverage you're adding here?